### DISCUSSION TITLE
Property 'navigateTo' does not exist on type '{ really long type }'

### DISCUSSION THREAD

**Discussion by BD103** [nuxt/nuxt#20819] [Questions] [answered]
Hello! The Vue linter in VSCode is show an error when using the following pattern:

```vue
<button @click="navigateTo('/')">Hello!</button>
```

It builds and runs fine, but underlines `navigateTo` and prints the following error message:

```
Property 'navigateTo' does not exist on type '{ $: ComponentInternalInstance; $data: {}; $props: { key?: string | number | symbol | undefined; ref?: VNodeRef | undefined; ref_for?: boolean | undefined; ... 8 more ...; style?: unknown; }; ... 10 more ...; $watch<T extends string | ((...args: any) => any)>(source: T, cb: T extends (...args: any) => infer R ? (arg...'.
```

This has just appeared recently, though I'm not sure if it's due to a dependency bump. I'll try to reproduce the issue on Stackblitz, though I don't know if it has the Intellisense required.

Any insight or solutions would be wonderful! I can link the source code if requested too. :)

# Specs

```
- Operating System: `Darwin`
- Node Version:     `v18.16.0`
- Nuxt Version:     `3.4.3`
- Nitro Version:    `2.4.0`
- Package Manager:  `pnpm@8.4.0`
- Builder:          `vite`
- User Config:      `components`, `modules`, `typescript`
- Runtime Modules:  `@huntersofbook/naive-ui-nuxt@0.7.1`
- Build Modules:    `-`
```

---

Edit: I was able to reproduce the bug locally using [this code](https://github.com/BD103/Nuxt-Bug-Reproduction), but I was not able to on Stackblitz (due to lack of a language server).

**Comment by kaikaibenkai** [comment]
Did you run 'npm run postinstall'?

---

Somthing in you code:
```diff
- <button @click="navigateTo('/')">Hello!</button>
+ <button @click="() => navigateTo('/')">Hello!</button>
```

**Comment by warflash** [comment, marked as answer]
> This has just appeared recently, though I'm not sure if it's due to a dependency bump.

Yes, this has to do with vue 3.3, you can check https://github.com/nuxt/nuxt/issues/20827 for more info. Should be resolved with nuxt 3.5
